Files/UserControls/LayoutModes/GenericFileBrowser.xaml.cs

```diff
@@ -260,8 +260,17 @@ private void AllView_PreparingCellForEdit(object sender, DataGridPreparingCellFo
 
             renamingTextBox = e.EditingElement as TextBox;
             renamingTextBox.Focus(FocusState.Programmatic); // Without this, cannot edit text box when renaming via right-click
-            renamingTextBox.Select(0, SelectedItem.ItemName.Length - extensionLength);
             renamingTextBox.TextChanged += TextBox_TextChanged;
+
+            if (App.AppSettings.ShowFileExtensions == true)
```

Comment:
I think it is possible to do the same with less code.
```csharp
            renamingTextBox = e.EditingElement as TextBox;
            renamingTextBox.Focus(FocusState.Programmatic); // Without this, cannot edit text box when renaming via right-click

            int selectedTextLength = SelectedItem.ItemName.Length;
            if (App.AppSettings.ShowFileExtensions)
            {
                selectedTextLength -= extensionLength;
            }
            renamingTextBox.Select(0, selectedTextLength);
            renamingTextBox.TextChanged += TextBox_TextChanged;
```
Same with GridViewBrowser.xaml.cs